3 new functions added to `y_weapondata`

DEntis-T opened the pull request.

DEntis-T committed:
`Weapon_GetType`

DEntis-T committed:
Weapon_GetType fix

DEntis-T committed:
`Weapon_GetDamage`

DEntis-T committed:
`Weapon_GetClipSize`

DEntis-T committed:
`Weapon_IsValid` before returning the data

DEntis-T commented:
OK, it's probably ready to be merged now, unless someone suggests something new on Discord

Y-Less commented on YSI_Game/y_weapondata/y_weapondata_entry.inc:
```diff
@@ -22,7 +22,6 @@ Legal:
 	Portions created by the Initial Developer are Copyright C 2011
 	the Initial Developer. All Rights Reserved.
-
 Contributors:
```
Why has this part changed?

Y-Less commented on YSI_Game/y_weapondata/y_weapondata_entry.inc:
```diff
@@ -87,7 +82,6 @@ Optional plugins:
 	88   `8b d8'   88 ,adPPPPP88 88 88       88      d8""""""""8b   88          88  
 	88    `888'    88 88,    ,88 88 88       88     d8'        `8b  88          88  
 	88     `8'     88 `"8bbdP"Y8 88 88       88    d8'          `8b 88          88  
-
```
All these mising blank lines in the header.

Y-Less commented on YSI_Game/y_weapondata/y_weapondata_entry.inc:
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
 	return WEAPONDATA_INVALID_RETURN;
 }
+
+#define WEAPON_TYPE_UNKNOWN				WeaponType:0
```
Usea an enum.

Y-Less commented on YSI_Game/y_weapondata/y_weapondata_entry.inc:
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
+#define WEAPON_TYPE_ASSAULTRIFLE 		WeaponType:3
+#define WEAPON_TYPE_SHOTGUN 			WeaponType:4
+#define WEAPON_TYPE_SUBMACHINEGUN 		WeaponType:5
+#define WEAPON_TYPE_RIFLE 				WeaponType:6
```
Should the type numbers not match the slot numbers?  Or if not ohw are they different (please document the difference)?

Y-Less commented on YSI_Game/y_weapondata/y_weapondata_entry.inc:
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
+#define WEAPON_TYPE_RIFLE 				WeaponType:6
+#define WEAPON_TYPE_PROJECTILE 			WeaponType:7
+#define WEAPON_TYPE_THROWABLE 			WeaponType:8
+#define WEAPON_TYPE_USEABLE				WeaponType:9
```
`USEABLE` is a terrible name, I know what you mean but all weapons are "useable".  Maybe `EQUIPMENT`?

Y-Less commented on YSI_Game/y_weapondata/y_weapondata_entry.inc:
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
+#define WEAPON_TYPE_SHOTGUN 			WeaponType:4
+#define WEAPON_TYPE_SUBMACHINEGUN 		WeaponType:5
+#define WEAPON_TYPE_RIFLE 				WeaponType:6
+#define WEAPON_TYPE_PROJECTILE 			WeaponType:7
```
`ARTILLERY`

DEntis-T committed:
Merge branch 'pawn-lang:5.x' into patch-4

DEntis-T committed:
Make Weapon Types an enum, and sync weapon type names with their slot names

Tested it!

```
@als() function OnGameModeInit(@rem(.level = BUG) : "lmao";)
{
    new weapid = 33;
    new WeaponType:type = Weapon_GetType(weapid);
    printf("Weapon type of weapon %i is %i", weapid, _:type);
    if(type == WEAPON_TYPE_LONG_RIFLE)
    {
        printf("weapon type is WEAPON_TYPE_LONG_RIFLE");
    }
    return @als() break;
}
```

DEntis-T committed:
Remove that little one

DEntis-T commented:
@Y-Less Solved the things you've posted, but something doesn't seem to work at a glance.
This code was working fine:
```pawn
new weapid = 33;
    new WeaponType:type = Weapon_GetType(weapid);
    printf("Weapon type of weapon %i is %i", weapid, _:type);
    if(type == WEAPON_TYPE_LONG_RIFLE)
    {
        printf("weapon type is WEAPON_TYPE_LONG_RIFLE");
    }
```
But suddenly it started to give a runtime error with an ID of 6 saying "Invalid instruction". I don't see anything invalid here

DEntis-T commented:
> @Y-Less Solved the things you've posted, but something doesn't seem to work at a glance. This code was working fine:
> 
> ```pawn
> new weapid = 33;
>     new WeaponType:type = Weapon_GetType(weapid);
>     printf("Weapon type of weapon %i is %i", weapid, _:type);
>     if(type == WEAPON_TYPE_LONG_RIFLE)
>     {
>         printf("weapon type is WEAPON_TYPE_LONG_RIFLE");
>     }
> ```
> 
> But suddenly it started to give a runtime error with an ID of 6 saying "Invalid instruction". I don't see anything invalid here

Maybe the problem is in this part:
```pawn
    if(type == WEAPON_TYPE_LONG_RIFLE)
    {
        printf("weapon type is WEAPON_TYPE_LONG_RIFLE");
    }
```
I'm currently really tired so I will leave this as it is now

DEntis-T commented:
Nevermind, I'll get it fixed

DEntis-T committed:
what was i doing

DEntis-T committed:
Merge branch 'pawn-lang:5.x' into patch-4

DEntis-T committed:
Merge branch 'pawn-lang:5.x' into patch-4

DEntis-T committed:
Embrace `switch` and `case`.

DEntis-T committed:
or just do a return...

DEntis-T committed:
Fix identation.

DEntis-T committed:
Fixes.

DEntis-T committed:
Update y_weapondata_entry.inc

DEntis-T committed:
`YSI_Weapon_IsValid__`

DEntis-T commented:
@Y-Less It works now.

DEntis-T committed:
Merge branch 'pawn-lang:5.x' into patch-4
